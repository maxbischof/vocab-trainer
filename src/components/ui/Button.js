import React from 'react'
import styled from 'styled-components'

export default function Button({ type, onClick, children, buttonstyle }) {
  const buttonStyles = {
    primary: {
      background: 'var(--primarybutton)',
    },
    secondary: {
      background: 'var(--secondarybutton)',
    },
  }

  return (
    <StyledButton
      type={type}
      onClick={onClick}
      buttonstyle={buttonStyles[buttonstyle]}
    >
      {children}
    </StyledButton>
  )
}

const StyledButton = styled.button`
  border: none;
  margin: 5px;
  background: ${(props) =>
    props.buttonstyle?.background || 'var(--primarybutton)'};
  color: white;
  border-radius: 10px;
  font-weight: 700;
  cursor: pointer;
  width: 100%;
  height: 36px;
  max-width: 238px;
`
