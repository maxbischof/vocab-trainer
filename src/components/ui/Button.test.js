import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

it('renders without button style correctly', () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders primary button style correctly', () => {
  const tree = renderer.create(<Button buttonstyle="primary" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders secondary button style correctly', () => {
  const tree = renderer.create(<Button buttonstyle="secondary" />).toJSON()
  expect(tree).toMatchSnapshot()
})
