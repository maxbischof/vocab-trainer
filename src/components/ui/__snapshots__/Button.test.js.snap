// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders primary button style correctly 1`] = `
<button
  className="sc-bdfBwQ hgDFIz"
/>
`;

exports[`renders secondary button style correctly 1`] = `
<button
  className="sc-bdfBwQ ftGbt"
/>
`;

exports[`renders without button style correctly 1`] = `
<button
  className="sc-bdfBwQ hgDFIz"
/>
`;
